src/Internal/Service/OrderService.php

```diff
@@ -254,6 +254,87 @@ public function attach_exchange_info_to_order( int $order_id, WC_Payments_API_Ch
 		$order->save_meta_data();
 	}
 
+	/**
+	 * Gets currently attached intent ID of the order.
+	 *
+	 * @param  int $order_id Order ID.
+	 *
+	 * @return string|null Intent ID for the order. Null if no intent ID attached to order.
+	 * @throws Order_Not_Found_Exception
+	 */
+	public function get_intent_id( int $order_id ): ?string {
+		$order     = $this->get_order( $order_id );
+		$intent_id = (string) $order->get_meta( '_intent_id', true );
+		if ( empty( $intent_id ) ) {
+			return null;
+		}
+		return $intent_id;
+	}
+
+	/**
+	 * Updates the order status from the given intent object.
+	 *
+	 * @param  int                                $order_id Order ID.
+	 * @param  WC_Payments_API_Abstract_Intention $intent Intent object.
+	 *
+	 * @return void
+	 * @throws Order_Not_Found_Exception
+	 */
+	public function update_order_status_from_intent( int $order_id, WC_Payments_API_Abstract_Intention $intent ): void {
+		$this->legacy_service->update_order_status_from_intent( $this->get_order( $order_id ), $intent );
```

Comment:
See the `update_order_from_successful_intent` method of this class. Let's at least here, create separate methods for separate outcomes. This one should likely make sure to update all needed meta as well.

Reply:
I am going to remove this method. It was a part of `check_payment_intent_attached_to_order_succeeded` for updating orders after the currenlty attached intent has one of successful ("authorized" is more correct, with our new glossary) statuses.

https://github.com/Automattic/woocommerce-payments/blob/9b414720c672ea0fcae13b64da3a2c0166e7efbe/includes/class-duplicate-payment-prevention-service.php#L117

Eventually, this action would be handed in a new post-processing state, which is currently a TODO https://github.com/Automattic/woocommerce-payments/pull/7450#discussion_r1362622441

Reply:
Removed here d09d1e03d7e60e3bfde8a2dc31b14abd4f88799f